packages/swing-store/src/transcriptStore.js

```diff
@@ -0,0 +1,443 @@
+            transcripts[vatID] = {};
+          }
+          transcripts[vatID][position] = item;
+        }
+      }
+    }
+    return transcripts;
+  }
+
+  /**
+   * Compute a cumulative hash that includes a new transcript item.  This is
+   * computed by hashing together the hash from the previous item in its span
+   * together with the new item's own text.
+   *
+   * @param {string} priorHash  The previous item's hash
+   * @param {string} item  The item itself
+   *
+   * @returns {string}  The hash of the combined parameters.
+   */
+  function computeItemHash(priorHash, item) {
+    const itemHash = createSHA256(item).finish();
+    return createSHA256(priorHash).add(itemHash).finish();
+  }
+
+  /**
+   * @type {string} Seed hash to use as the prior hash when computing the hash
+   * of the very first item in a span, since it has no prior item to draw upon.
+   */
+  const initialHash = createSHA256('start of transcript span').finish();
+
+  const sqlWriteSpan = db.prepare(`
+    INSERT INTO transcriptSpans
+      (vatID, startPos, endPos, hash, isCurrent)
+    VALUES (?, ?, ?, ?, ?)
+  `);
+
+  /**
+   * Start a new transcript for a given vat
+   *
+   * @param {string} vatID  The vat whose transcript this shall be
+   */
+  function initTranscript(vatID) {
+    ensureTxn();
+    sqlWriteSpan.run(vatID, 0, 0, initialHash, 1);
+  }
+
+  const sqlGetCurrentSpanBounds = db.prepare(`
+    SELECT startPos, endPos, hash
+    FROM transcriptSpans
+    WHERE vatID = ? AND isCurrent = 1
+  `);
+
+  /**
+   * Obtain the bounds and other metadata for a vat's current transcript span.
+   *
+   * @param {string} vatID  The vat in question
+   *
+   * @returns {{startPos: number, endPos: number, hash: string}}
+   */
+  function getCurrentSpanBounds(vatID) {
+    const bounds = sqlGetCurrentSpanBounds.get(vatID);
+    assert(bounds, `no current transcript for ${vatID}`);
+    return bounds;
+  }
+
+  function spanArtifactName(rec) {
+    return `transcript.${rec.vatID}.${rec.startPos}.${rec.endPos}`;
+  }
+
+  function historicSpanMetadataKey(rec) {
+    if (rec.isCurrent) {
+      return `transcript.${rec.vatID}.current`;
+    } else {
+      return `transcript.${rec.vatID}.${rec.startPos}`;
+    }
+  }
+
+  function currentSpanMetadataKey(rec) {
+    return `transcript.${rec.vatID}.current`;
+  }
+
+  function spanRec(vatID, startPos, endPos, hash, isCurrent) {
+    return { vatID, startPos, endPos, hash, isCurrent };
+  }
+
+  const sqlEndCurrentSpan = db.prepare(`
+    UPDATE transcriptSpans
+    SET isCurrent = 0
+    WHERE isCurrent = 1 AND vatID = ?
+  `);
+
+  /**
+   * End the current transcript span for a vat and start a new one.
+   *
+   * @param {string} vatID  The vat whose transcript is to rollover to a new
+   *    span.
+   */
+  function rolloverSpan(vatID) {
+    ensureTxn();
+    const { hash, startPos, endPos } = getCurrentSpanBounds(vatID);
+    const rec = spanRec(vatID, startPos, endPos, hash);
```

Comment:
Missing `false` here
```suggestion
    const rec = spanRec(vatID, startPos, endPos, hash, false);
```